File: INF/statgrad2/17/17.cpp
#include <iostream>
#include <string>
#include <fstream>
#include <vector>

int main() {
	std::ifstream f;
	f.open("17.txt");
	int x, sum = 0, count = 0, k = 0, max = 0;
	double average;
	std::vector<int> v;
	while(!f.eof()) {
		f >> x;
		if(x % 2 != 0) {
			sum += x;
			count += 1;
		}
		v.push_back(x);
	}
	average = sum / count;
	for(int i = 1; i < v.size(); ++i) {
		if(v[i] % 5 == 0 || v[i - 1] % 5 == 0) {
			if(v[i] % 5 == 0) {
				if(v[i - 1] < average) {
					k++;
					if(v[i - 1] + v[i] > max) 
						max = v[i - 1] + v[i];
				}
			}else if(v[i - 1] % 5 == 0) {
				if(v[i] < average) {
					k++;
					if(v[i - 1] + v[i] > max) 
						max = v[i - 1] + v[i];
				}
			}
		}
	}
	std::cout << k << ' ' << max;
	f.close();
	return 0;
}

File: skopintseva_may/17.py
def g(n):
    k = 0
    while(n):
        k += 1
        n //= 10
    if k == 2:
        return True
    return False

with open("17.txt") as f: 
    count = 0
    lst = []
    mn = 10000
    mx = 0
    for j in f:
        i = int(j)
        lst.append(i)
        if i % 17 == 0 and i < mn:
            mn = i
    for i in range(len(lst) - 1):
        a = lst[i]
        b = lst[i + 1]
        if (g(a) or g(b)):
            if lst[i] + lst[i + 1] < mn:
                count += 1
                if a + b > mx:
                    mx = a + b
    print(count, mx)
